ocaml/xapi/xapi_vbd.ml

```diff
@@ -47,10 +77,24 @@ let plug ~__context ~self =
     Storage_access.attach_and_activate ~__context ~vbd:self ~domid
       (fun attach_info ->
          let params = attach_info.Storage_interface.params in
-         let prefix = "/dev/" in
-         let prefix_len = String.length prefix in
-         let path = String.sub params prefix_len (String.length params - prefix_len) in
-         Db.VBD.set_device ~__context ~self ~value:path;
+         let device_path =
+           let nbd_prefix = "hack|nbd:unix:" in
+           let is_nbd = String.startswith nbd_prefix params in
```

Comment:
I should use `Astring` here too :/

Reply:
As you wish. Xapi is still not ported to Astring at all, feel free to use `Xstringext` here

Reply:
I've done it now